_posts/2021-05-18-My-First-Steps-in-OpenSearch-Plugins.markdown

```diff
@@ -0,0 +1,618 @@
+
+Turns out, this may be true for *small-scale* projects, but for complex ones like OpenSearch this isn't the case. While you may fix a minor bug, you're not likely to know where to begin in order to fix a moderate bug or design a new feature. Searching for that entry point via the `main()` entrance mostly feels like wandering around an [M.C. Escher](https://en.wikipedia.org/wiki/Relativity_(M._C._Escher)) print.
+
+However, some projects have an entrance unlike any other - a **Pluggable Architecture**. 
+
+
+The Pluggable-Architecture enables us to have a hands on experience with a logical sub section of a large-scale project. This effect makes such an immense project more approachable despite its overall size.
+
+
+In the following blog post, I will take you on a journey through my thought process and conclusions from building *my* first OpenSearch plugin. We will build a simple REST plugin together, install it into OpenSearch, and run it. By the end, I hope to convince you that it is a great starting point for you gaining hands-on experience required for joining the OpenSearch community of contributors, and a great starting point for the project itself.
+
+
+---
+- [My First Steps in OpenSearch Plugins](#my-first-steps-in-opensearch-plugins)
+    - [Taking the leap, not the _plunge_](#taking-the-leap-not-the-plunge)
+  - [What are OpenSearch Plugins?](#what-are-opensearch-plugins)
+    - [1. Core Plugins](#1-core-plugins)
+    - [2. Community Contributed Plugins](#2-community-contributed-plugins)
+- [Building a Simple OpenSearch REST Plugin](#building-a-simple-opensearch-rest-plugin)
+    - [Before we start - make sure you have these environment dependencies](#before-we-start---make-sure-you-have-these-environment-dependencies)
+    - [Defining Our Plugin Requirements for OpenSearch](#defining-our-plugin-requirements-for-opensearch)
+  - [The `build.gradle` of Our Plugin](#the-buildgradle-of-our-plugin)
+  - [Implementing the Plugin Code](#implementing-the-plugin-code)
+  - [OpenSearch Plugin Interfaces](#opensearch-plugin-interfaces)
+  - [Implementing Plugin logic](#implementing-plugin-logic)
+  - [Logic separation](#logic-separation)
+  - [Testing](#testing)
+- [Trying out our new plugin](#trying-out-our-new-plugin)
+  - [Building the plugin project](#building-the-plugin-project)
+  - [Installing the plugin](#installing-the-plugin)
+  - [Running OpenSearch with our new plugin](#running-opensearch-with-our-new-plugin)
+  - [Uninstalling the plugin](#uninstalling-the-plugin)
+- [Key takeaways from creating my first plugin](#key-takeaways-from-creating-my-first-plugin)
+
+## What are OpenSearch Plugins? 
+
+Software plugins are, in essence, stand alone components __add features and capabilities__ to an existing core program. 
```

Comment:
I think there might be a missing "that" in this sentence. Does the below look right?

Software plugins are, in essence, stand alone components that add features and capabilities to an existing core program.